### DISCUSSION TITLE
Troubles in boot

### DISCUSSION THREAD

**Discussion by Shlepsss** [CosmosOS/Cosmos#2148] [Q&A]
![image](https://user-images.githubusercontent.com/79642857/156374106-ab5603d0-42eb-49a3-a59e-32889c733f8f.png)
Please help, system don't boot! System stop at this moment!!!

**Comment by MishaTy** [comment]
Can you send your code?

**Reply by Shlepsss** [reply]
using System;
using sus = Cosmos.System;
using hw = Cosmos.HAL;
using scr = koteos.Kernel;
using Cosmos.Core;
using mouse = Cosmos.System.MouseManager;

namespace koteos
{
    public class Kernel : sus.Kernel
    {
        public static string num1;
        public static string num2;
        //public static bool iswindowcalc;
        public static bool iswindowram;
        public Cosmos.HAL.MouseBase mb;
        public static string chossss;
        public static Cosmos.System.Graphics.VBECanvas screen;
        public static bool whiletrue;
        public static System.Drawing.Color iscolor;
        public static System.Drawing.Color iscolorpurple;
        public static sus.Graphics.Pen penn;
        protected override void BeforeRun()
        {
            whiletrue = true;
            iswindowram = false;
            var fs = new sus.FileSystem.CosmosVFS();
            string current_directory = "0:\\";
            sus.FileSystem.VFS.VFSManager.RegisterVFS(fs);
            //iswindowcalc = false;
        }

        public static bool CheackClick(int x, int y, int w, int h)
        {
            if (sus.MouseManager.X <= x + w && sus.MouseManager.X >= x)
            {
                if (sus.MouseManager.Y <= y + h && sus.MouseManager.Y >= y)
                {
                    if (sus.MouseManager.MouseState == sus.MouseState.Left)
                    {
                        return true;
                    }
                }
            }
            return false;
        }
        public static void lookram()
        {
            penn.Color = System.Drawing.Color.Blue;
            screen.DrawFilledRectangle(penn, 50, 90, 100, 30);
            txtxtx.FontDrawer.WriteText("+++++++++++++", 56, 90, 63);
            txtxtx.FontDrawer.WriteText("/Console mode/", 56, 100, 63);
            txtxtx.FontDrawer.WriteText("=============", 56, 109, 63);
            penn.Color = System.Drawing.Color.Black;

            //   ] - мозайка   [ - квадрат полный   - - решётка наискосок   _ - квадрат с решёткой наискосок   | - квадрат неполный   = - верхный дефис   + - нижний дефис   / - полоска по вертикали

            //var di = new DriveInfo("0");
        }
        public static void panel()
        {
            penn.Color = System.Drawing.Color.Violet;
            screen.DrawFilledRectangle(penn, 280, 0, 40, 200);
            penn.Color = System.Drawing.Color.Black;
        }
        public static void antipanel()
        {
            penn.Color = System.Drawing.Color.Black;
            screen.DrawFilledRectangle(penn, 280, 0, 40, 200);
            penn.Color = System.Drawing.Color.Black;
        }
        /*public static void calc(string num1, string num2)
        {
            DrawFilledRectangle(60, 70, 150, 45, 3);
            DrawFilledRectangle(65, 75, 140, 10, 8);
            DrawFilledRectangle(65, 100, 140, 10, 8);
            DrawFilledRectangle(65, 85, 20, 10, 7);
            DrawFilledRectangle(90, 85, 20, 10, 7);
            DrawFilledRectangle(115, 85, 20, 10, 7);
            DrawFilledRectangle(140, 85, 20, 10, 7);
            txtxtx.FontDrawer.WriteText("PL  MI  DE  MU", 70, 86, 0);
            float num11 = Convert.ToSingle(num1);
            float num22 = Convert.ToSingle(num2);
            if (CheackClick(65, 85, 20, 10))
            {
                DrawFilledRectangle(65, 100, 140, 10, 8);
                float ans = num11 + num22;
                string anss = Convert.ToString(ans);
                txtxtx.FontDrawer.WriteText(anss, 66, 101, 63);
            }
            if (CheackClick(90, 85, 20, 10))
            {
                DrawFilledRectangle(65, 100, 140, 10, 8);
                float ans = num11 - num22;
                string anss = Convert.ToString(ans);
                txtxtx.FontDrawer.WriteText(anss, 66, 101, 63);
            }
            if (CheackClick(115, 85, 20, 10))
            {
                DrawFilledRectangle(65, 100, 140, 10, 8);
                float ans = num11 / num22;
                string anss = Convert.ToString(ans);
                txtxtx.FontDrawer.WriteText(anss, 66, 101, 63);
            }
            if (CheackClick(140, 85, 20, 10))
            {
                DrawFilledRectangle(65, 100, 140, 10, 8);
                float ans = num11 * num22;
                string anss = Convert.ToString(ans);
                txtxtx.FontDrawer.WriteText(anss, 66, 101, 63);
            }
        }*/
        protected override void Run()
        {
            try
            {
                iswindowram = false;
                scr.screen.Disable();
                whiletrue = true;
                Console.Clear();
                Console.WriteLine("FLOOPOS CONSOLE.");
                while (whiletrue)
                {
                    var txt = Console.ReadLine();
                    switch (txt)
                    {
                        case "exit":
                            sus.Power.Shutdown();
                            break;
                        case "help":
                            Console.WriteLine("clear - clear the console");
                            Console.WriteLine("calc - calculator");
                            Console.WriteLine("time - now time");
                            Console.WriteLine("help - help");
                            Console.WriteLine("proc - find the percentage of the number");
                            Console.WriteLine("reboot - reboot");
                            Console.WriteLine("exit - shutdown");
                            Console.WriteLine("settings - settings");
                            Console.WriteLine("standby - standby mode");
                            Console.WriteLine("3d - Look at 3D Images!");
                            Console.WriteLine("images - Look at 2D Images!");
                            Console.WriteLine("game - play game 'guess the number'");
                            Console.WriteLine("mtest - mathe test!");
                            Console.WriteLine("mel - lesson the music");
                            Console.WriteLine("backtog - back to graphics mode");
                            Console.WriteLine("");
                            break;
                        case "reboot":
                            sus.Power.Reboot();
                            break;
                        case "clear":
                            Console.Clear();
                            break;
                        case "calc":
                            double a, b;

                            Console.WriteLine("");
                            try
                            {
                                Console.Write("Enter first number: ");

                                a = double.Parse(Console.ReadLine());
                                Console.WriteLine("");
                                Console.Write("Enter second number: ");

                                b = double.Parse(Console.ReadLine());
                            }
                            catch
                            {
                                Console.WriteLine("You are an idiot?");
                                break;
                            }
                            Console.WriteLine("Choose an option:");
                            Console.WriteLine("a - add");
                            Console.WriteLine("s - subtract");
                            Console.WriteLine("m - multiply");
                            Console.WriteLine("d - devide");
                            Console.Write("You option:");

                            switch (Console.ReadLine())
                            {
                                case "a":
                                    Console.WriteLine(a + b);
                                    Console.WriteLine("");
                                    break;
                                case "s":
                                    Console.WriteLine(a - b);
                                    Console.WriteLine("");
                                    break;
                                case "m":
                                    Console.WriteLine(a * b);
                                    Console.WriteLine("");
                                    break;
                                case "d":
                                    Console.WriteLine(a / b);
                                    Console.WriteLine("");
                                    break;
                                default:
                                    Console.WriteLine("ERROR: UNKNOWN OPTION");
                                    Console.WriteLine("");
                                    break;
                            }
                            break;
                        case "time":
                            Console.WriteLine(DateTime.Now);
                            Console.WriteLine("");
                            break;
                        case "proc":
                            double sd, sf;
                            Console.WriteLine("");
                            Console.WriteLine("You can find the percentage of the number");
                            Console.WriteLine("");
                            try
                            {
                                Console.Write("Enter percentages: ");
                                sd = double.Parse(Console.ReadLine());
                                Console.Write("Enter number: ");
                                sf = double.Parse(Console.ReadLine());
                            }
                            catch
                            {
                                Console.WriteLine("You are an idiot?");
                                break;
                            }
                            double r = sf / 100;
                            double res = r * sd;
                            Console.WriteLine("Result: " + res);
                            break;
                        case "settings":
                            Console.WriteLine("theme - set the existing color set");
                            Console.WriteLine("bg - set background color");
                            Console.WriteLine("tc - set the text color");
                            string setttt = Console.ReadLine();
                            switch (setttt)
                            {
                                case "theme":
                                    Console.WriteLine("Choose a theme: dark, light, mc (multicolored), sd (standart), flopoos(author's favorite theme)");
                                    Console.WriteLine("");
                                    Console.Write("You choose: ");
                                    string cs = Console.ReadLine();
                                    switch (cs)
                                    {
                                        case "dark":
                                            Console.ForegroundColor = ConsoleColor.Gray;
                                            Console.BackgroundColor = ConsoleColor.Black;
                                            Console.Clear();
                                            break;
                                        case "light":
                                            Console.ForegroundColor = ConsoleColor.White;
                                            Console.BackgroundColor = ConsoleColor.Gray;
                                            Console.Clear();
                                            break;
                                        case "mc":
                                            Console.ForegroundColor = ConsoleColor.DarkBlue;
                                            Console.BackgroundColor = ConsoleColor.Green;
                                            Console.Clear();
                                            break;
                                        case "bios":
                                            Console.ResetColor();
                                            Console.BackgroundColor = ConsoleColor.Blue;
                                            Console.Clear();
                                            break;
                                        case "flopoos":
                                            Console.ResetColor();
                                            Console.BackgroundColor = ConsoleColor.Black;
                                            Console.ForegroundColor = ConsoleColor.DarkRed;
                                            Console.Clear();
                                            break;
                                    }
                                    break;
                                case "bg":
                                    Console.WriteLine("please set the bg color... (g - green, r - red, b - blue, bl - black, y - orange, c - cyan, m - magenta, gr - gray)");
                                    string bgc = Console.ReadLine();
                                    if (bgc == "g")
                                    {
                                        Console.BackgroundColor = ConsoleColor.Green;
                                        Console.Clear();
                                    }
                                    if (bgc == "r")
                                    {
                                        Console.BackgroundColor = ConsoleColor.Red;
                                        Console.Clear();
                                    }
                                    if (bgc == "b")
                                    {
                                        Console.BackgroundColor = ConsoleColor.Blue;
                                        Console.Clear();
                                    }
                                    if (bgc == "bl")
                                    {
                                        Console.BackgroundColor = ConsoleColor.Black;
                                        Console.Clear();
                                    }
                                    if (bgc == "y")
                                    {
                                        Console.BackgroundColor = ConsoleColor.Yellow;
                                        Console.Clear();
                                    }
                                    if (bgc == "c")
                                    {
                                        Console.BackgroundColor = ConsoleColor.Cyan;
                                        Console.Clear();
                                    }
                                    if (bgc == "m")
                                    {
                                        Console.BackgroundColor = ConsoleColor.Magenta;
                                        Console.Clear();
                                    }
                                    if (bgc == "gr")
                                    {
                                        Console.BackgroundColor = ConsoleColor.Gray;
                                        Console.Clear();
                                    }
                                    break;
                                case "tc":
                                    Console.WriteLine("Please, set the text color... (g - green, r - red, b - blue, w - white, y - yellow, c - cyan, dg - dark green, m - magenta, gr - gray)");
                                    string clr = Console.ReadLine();
                                    if (clr == "g")
                                    {
                                        Console.ForegroundColor = ConsoleColor.Green;
                                        Console.Clear();
                                    }
                                    if (clr == "r")
                                    {
                                        Console.ForegroundColor = ConsoleColor.Red;
                                        Console.Clear();
                                    }
                                    if (clr == "b")
                                    {
                                        Console.ForegroundColor = ConsoleColor.Blue;
                                        Console.Clear();
                                    }
                                    if (clr == "w")
                                    {
                                        Console.ForegroundColor = ConsoleColor.White;
                                        Console.Clear();
                                    }
                                    if (clr == "y")
                                    {
                                        Console.ForegroundColor = ConsoleColor.Yellow;
                                        Console.Clear();
                                    }
                                    if (clr == "c")
                                    {
                                        Console.ForegroundColor = ConsoleColor.Cyan;
                                        Console.Clear();
                                    }
                                    if (clr == "dg")
                                    {
                                        Console.ForegroundColor = ConsoleColor.DarkGreen;
                                        Console.Clear();
                                    }
                                    if (clr == "m")
                                    {
                                        Console.ForegroundColor = ConsoleColor.Magenta;
                                        Console.Clear();
                                    }
                                    if (clr == "gr")
                                    {
                                        Console.ForegroundColor = ConsoleColor.Gray;
                                        Console.Clear();
                                    }
                                    if (clr == "bl")
                                    {
                                        Console.ForegroundColor = ConsoleColor.Black;
                                        Console.Clear();
                                    }
                                    if (clr == "drr")
                                    {
                                        Console.ForegroundColor = ConsoleColor.DarkRed;
                                        Console.Clear();
                                    }
                                    break;
                            }
                            break;
                        case "standby":
                            Console.BackgroundColor = ConsoleColor.Black;
                            Console.ForegroundColor = ConsoleColor.Gray;
                            Console.Clear();
                            Console.ReadKey();
                            Console.ResetColor();
                            Console.BackgroundColor = ConsoleColor.Blue;
                            Console.Clear();
                            break;
                        case "images":
                            Console.WriteLine("");
                            Console.WriteLine("1. Tree");
                            Console.WriteLine("2. Sunrice");
                            Console.WriteLine("3. Bunny");
                            Console.WriteLine("4. Cat");
                            Console.WriteLine("5. Beach");
                            Console.WriteLine("");
                            Console.Write("Enter number of image: ");
                            string ImgCh = Console.ReadLine();
                            switch (ImgCh)
                            {
                                case "1":
                                    Console.WriteLine("");
                                    Console.WriteLine("     @@@     ");
                                    Console.WriteLine("    @@@@@    ");
                                    Console.WriteLine("   @@@@@@@   ");
                                    Console.WriteLine("  @@@@@@@@@  ");
                                    Console.WriteLine("   @@@@@@@   ");
                                    Console.WriteLine("     ###  __ ");
                                    Console.WriteLine("  ___### /   ");
                                    Console.WriteLine("     ###/    ");
                                    Console.WriteLine("   --###     ");
                                    Console.WriteLine("     ###     ");
                                    Console.WriteLine("#############");
                                    Console.WriteLine("#############");
                                    Console.WriteLine("");
                                    break;
                                case "2":
                                    Console.WriteLine("");
                                    Console.WriteLine("...,,,:::+++***???##?***+++:::,,,...");
                                    Console.WriteLine("..,,:::+++***???S####S???***+++:::,,..");
                                    Console.WriteLine("..,,::+++***???S##@@##S???***+++::,,..");
                                    Console.WriteLine("..,,::++***???SS##@@##SS???***++::,,..");
                                    Console.WriteLine("..,,::++**???%SS##@@##SS%???**++::,,..");
                                    Console.WriteLine("..,,::++**??%%SS##@@##SS%%??**++::,,..");
                                    Console.WriteLine("@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@");
                                    Console.WriteLine("");
                                    break;
                                case "3":
                                    Console.WriteLine("");
                                    Console.WriteLine("  /]__/]");
                                    Console.WriteLine(" ( @  @)");
                                    Console.WriteLine(" (   ^ )");
                                    Console.WriteLine(" (  >##)");
                                    Console.WriteLine(" (_____)");
                                    Console.WriteLine("  !   !");
                                    Console.WriteLine("");
                                    break;
                                case "4":
                                    Console.WriteLine("");
                                    Console.WriteLine("            /]___/]");
                                    Console.WriteLine("     ______/ @  @ |");
                                    Console.WriteLine("    /      ____^__|");
                                    Console.WriteLine(" ---|  __ /        ");
                                    Console.WriteLine("    |_| |_|        ");
                                    Console.WriteLine("");
                                    break;
                                case "5":
                                    Console.WriteLine("");
                                    Console.WriteLine("     =#@@@@/@@= ");
                                    Console.WriteLine("           |    ");
                                    Console.WriteLine("           /    ");
                                    Console.WriteLine("__________/_____");
                                    Console.WriteLine("_________|______");
                                    Console.WriteLine("");
                                    break;
                                default:
                                    Console.WriteLine("Unknown Image");
                                    break;
                            }
                            break;
                        case "3d":
                            Console.WriteLine("");
                            Console.WriteLine("1. 3D Cube");
                            Console.WriteLine("2. 3D Letter 'T'");
                            Console.WriteLine("");
                            Console.Write("Enter number of image: ");
                            string treeD = Console.ReadLine();
                            switch (treeD)
                            {
                                case "1":
                                    Console.WriteLine("");
                                    Console.WriteLine("     ___________ ");
                                    Console.WriteLine("    /         / | ");
                                    Console.WriteLine("   /         /  | ");
                                    Console.WriteLine("  /         /   | ");
                                    Console.WriteLine(" -----------    | ");
                                    Console.WriteLine(" |         |    / ");
                                    Console.WriteLine(" |         |   / ");
                                    Console.WriteLine(" |         |  / ");
                                    Console.WriteLine(" ------------- ");
                                    Console.WriteLine("");
                                    break;
                                case "2":
                                    Console.WriteLine("");
                                    Console.WriteLine("  _________");
                                    Console.WriteLine(" /________/|");
                                    Console.WriteLine(" |________|/ ");
                                    Console.WriteLine("   | | |");
                                    Console.WriteLine("   | | |");
                                    Console.WriteLine("   | | |");
                                    Console.WriteLine("   | | |");
                                    Console.WriteLine("   | | |");
                                    Console.WriteLine("   |_|/");
                                    Console.WriteLine("");
                                    break;
                                default:
                                    Console.WriteLine("Unknown 3D Image");
                                    break;
                            }
                            break;
                        case "game":
                            Console.Write("Choose mode (1. easy 2. hard) enter number: ");
                            string chs = Console.ReadLine();
                            switch (chs)
                            {
                                case "1":
                                    Random rand = new Random();
                                    int ina = rand.Next(25);
                                    Console.WriteLine("I made a number from 0 to 25, find this number!");

                                    while (true)
                                    {
                                        if (ina < 15) Console.WriteLine("number < 15");
                                        else Console.WriteLine("number > or = 15");

                                        try
                                        {
                                            int x = Convert.ToInt32(Console.ReadLine());
                                            if (ina == x)
                                            {
                                                Console.WriteLine("Good! You Win!");
                                                break;
                                            }
                                            else
                                            {
                                                Console.WriteLine("No!");
                                                continue;
                                            }
                                        }
                                        catch
                                        {
                                            Console.WriteLine("You are an idiot?");
                                        }

                                    }
                                    break;
                                case "2":
                                    Random rando = new Random();
                                    int inaa = rando.Next(75);
                                    Console.WriteLine("I made a number from 0 to 75, find this number!");

                                    while (true)
                                    {
                                        if (inaa < 35) Console.WriteLine("number < 35");
                                        else Console.WriteLine("number > or = 35");
                                        try
                                        {
                                            int x = Convert.ToInt32(Console.ReadLine());
                                            if (inaa == x)
                                            {
                                                Console.WriteLine("Good! You Win!");
                                                break;
                                            }
                                            else
                                            {
                                                Console.WriteLine("No!");
                                                continue;
                                            }
                                        }
                                        catch
                                        {
                                            Console.WriteLine("You are an idiot?");
                                        }
                                    }
                                    break;
                                default:
                                    Console.WriteLine("Unknown Mode!");
                                    break;
                            }
                            break;
                        case "mtest":
                            try
                            {
                                Random shoo = new Random();
                                Random sho = new Random();
                                int sdf = shoo.Next(9999);
                                int fwt = sho.Next(8346);
                                int answer = sdf + fwt;
                                Console.Write("" + sdf + " + " + fwt + " = ");
                                while (true)
                                {
                                    string ya = Console.ReadLine();
                                    int yaa = Convert.ToInt32(ya);
                                    if (yaa == answer)
                                    {
                                        Console.WriteLine("Good!");
                                        break;
                                    }
                                    else
                                    {
                                        Console.WriteLine("Wrong!");
                                        continue;
                                    }
                                }
                            }
                            catch (Exception)
                            {
                                Console.WriteLine("ERROR!!!");
                                break;
                            }
                            break;
                        case "mel":
                            Console.WriteLine("Do you want lesson the music? TIP: IF YOU DONT HEAR THE MUSIC, CHECK BIOS FOR FUNCTION 'ENABLE BEEP' OR 'ENABLE SPEAKER' OR etc... IF LAST HINT DONT HELP, CHECK YOU PC FOR HAS SPEAKER, IF YOU PC DONT HAVE A SPEAKER FORGET THIS COMMAND :( SORRY!");
                            Console.WriteLine("Choose Music: 1 - Happy Birthday 2 - Mission Impossible 3 - mobilnik");
                            switch (Console.ReadKey(true).Key)
                            {
                                case ConsoleKey.D1:
                                    melody.mel.HappyBirthday();
                                    break;
                                case ConsoleKey.D2:
                                    melody.mel.MissionImpossible();
                                    break;
                                case ConsoleKey.D3:
                                    melody.mel.mobilnik();
                                    break;
                                default:
                                    break;
                            }
                            break;
                        case "graph":
                            whiletrue = false;
                            break;
                        default:
                            Console.WriteLine("Unknown Command!");
                            sus.PCSpeaker.Beep();
                            break;
                    }
                    if (whiletrue == false)
                    {
                        scr.screen.Display();
                        int mont;
                        mont = DateTime.Now.Month;
                        string montt = "";
                        Cosmos.System.MouseManager.ScreenHeight = 200;
                        Cosmos.System.MouseManager.ScreenWidth = 317;
                        switch (mont)
                        {
                            case 01:
                                montt = "January";
                                break;
                            case 02:
                                montt = "February";
                                break;
                            case 03:
                                montt = "March";
                                break;
                            case 04:
                                montt = "April";
                                break;
                            case 05:
                                montt = "May";
                                break;
                            case 06:
                                montt = "June";
                                break;
                            case 07:
                                montt = "July";
                                break;
                            case 08:
                                montt = "August";
                                break;
                            case 09:
                                montt = "September :(";
                                break;
                            case 10:
                                montt = "October";
                                break;
                            case 11:
                                montt = "November";
                                break;
                            case 12:
                                montt = "December";
                                break;
                        }
                        while (true)
                        {
                            penn.Color = System.Drawing.Color.DarkRed;
                            screen.DrawFilledRectangle(penn, 5, 140, 25, 25);
                            txtxtx.FontDrawer.WriteText(DateTime.Now.Hour.ToString() + ":" + DateTime.Now.Minute.ToString() + ":" + DateTime.Now.Second.ToString(), 235, 10, 63);
                            txtxtx.FontDrawer.WriteText(montt, 235, 20, 36);
                            txtxtx.FontDrawer.WriteText(DateTime.Now.Day.ToString() + "th " + DateTime.Now.Year.ToString(), 235, 30, 36);
                            txtxtx.FontDrawer.WriteText("]Console]", 45, 190, 36);
                            penn.Color = System.Drawing.Color.Red;
                            screen.DrawFilledRectangle(penn, 5, 170, 25, 25);
                            uint Y = Cosmos.System.MouseManager.Y;
                            uint X = Cosmos.System.MouseManager.X;
                            penn.Color = System.Drawing.Color.White;
                            screen.DrawPoint(penn, X, Y);
                            screen.DrawPoint(penn, X + 1, Y);
                            screen.DrawPoint(penn, X + 2, Y);
                            screen.DrawPoint(penn, X, Y + 1);
                            screen.DrawPoint(penn, X, Y + 2);
                            screen.DrawPoint(penn, X + 1, Y + 1);
                            screen.DrawPoint(penn, X + 2, Y + 2);
                            screen.DrawPoint(penn, X + 3, Y + 3);
                            if (sus.MouseManager.X >= 280)
                            {
                                iscolorpurple = screen.GetPointColor(300, 150);
                                if (iscolorpurple == System.Drawing.Color.Black)
                                {
                                    panel();
                                }
                            }
                            if (sus.MouseManager.X <= 279)
                            {
                                iscolor = screen.GetPointColor(290, 10);
                                if (iscolor == System.Drawing.Color.Violet)
                                {
                                    antipanel();
                                }
                            }
                            if (CheackClick(5, 170, 25, 25))
                            {
                                sus.Power.Shutdown();
                            }
                            if (CheackClick(5, 140, 25, 25))
                            {
                                sus.Power.Reboot();
                            }
                            if (CheackClick(34, 190, 45, 10))
                            {
                                if (iswindowram == false)
                                {
                                    lookram();
                                    iswindowram = true;
                                }
                            }
                            if (CheackClick(0, 121, 319, 78))
                            {
                                penn.Color = System.Drawing.Color.Black;
                                screen.DrawFilledRectangle(penn, 50, 90, 100, 30);
                                iswindowram = false;
                                screen.Disable();
                                penn.Color = System.Drawing.Color.Black;
                            }
                            /*if (CheackClick(64, 190, 38, 10))
                            {
                                if (iswindowcalc == false)
                                {
                                    iswindowcalc = true;
                                    num1 = Console.ReadLine();
                                    num2 = Console.ReadLine();
                                    calc(num1, num2);
                                }
                            }
                            if (iswindowcalc == true)
                            {
                                calc(num1, num2);
                                if (CheackClick(60, 70, 150, 45))
                                {
                                    DrawFilledRectangle(60, 70, 150, 45, 0);
                                    iswindowcalc = false;
                                }
                            }*/
                            break;
                        }
                    }
                }
            }
            catch (Exception)
            {
                sus.Power.Reboot();
            }
            //ReDraw();
            
        }
    }
}

**Reply by ghost** [reply]
@koteProgrammist did you use a userkit or devkit?

**Reply by Shlepsss** [reply]
userkit

> @koteProgrammist did you use a userkit or devkit?

userkit

**Reply by quajak** [reply]
Is the screen enabled before this is called `scr.screen.Disable();`? It might be a good idea to simplify the code until it works to figure out what the issue is.
